test/unit/eigensolver/test_gen_eigensolver.cpp

```diff
@@ -9,7 +9,6 @@
 //
 
 #include <fstream>
-#include <numbers>
```

Comment:
Do we have any idea how this was not caught by CI?

Reply:
I just realised that using an old compiler (namely, the good old gcc@9 on Daint).

> Header available in libstdc++ but exposes nothing in C++17 mode?

Apparently, yes. Some evidence:
- https://godbolt.org/z/zoWbjoG58
- [`<numbers>` in GCC](https://github.com/gcc-mirror/gcc/blob/be8d1a358e3abc50c14a1d7b1cfee82fe6f6aa3c/libstdc%2B%2B-v3/include/std/numbers#L39)
- [`<numbers>` in LLVM](https://github.com/llvm/llvm-project/blob/c1ad064dd3b9a1bf503bb8cd6d0d0418a05d10e8/libcxx/include/numbers#L65)

Reply:
Just to unlink it from this PR, do we want to open an issue and keep track of that? or we are not even interested since there seems no way of checking that?

Reply:
As quickly discussed with @rasolca, we are not going to keep track of this.

CI didn't realise it because by using sufficiently new compilers the header is available anyway, but since it was actually unused, it was not triggering any compilation error.

So, it should either be
- compiler error because not available in the cpp standard used
- an unused header

The former is enforced by the compiler, the latter might be detected by a more thorough check about actually unused headers.

Reply:
Thanks for looking into this! Yeah, I suppose as long as we don't have a proper include-what-you-use check that also removes includes we can't do much about these kind of issues.